miio/viomivacuum.py

```diff
@@ -227,96 +223,265 @@ def error(self) -> Optional[str]:
     @property
     def battery(self) -> int:
         """Battery in percentage."""
-        return self.data["battary_life"]
+        return self._data["battary_life"]
 
     @property
     def bin_type(self) -> ViomiBinType:
         """Type of the inserted bin."""
-        return ViomiBinType(self.data["box_type"])
+        return ViomiBinType(self._data["box_type"])
 
     @property
     def clean_time(self) -> timedelta:
         """Cleaning time."""
-        return pretty_seconds(self.data["s_time"])
+        return pretty_seconds(self._data["s_time"])
 
     @property
     def clean_area(self) -> float:
         """Cleaned area in square meters."""
-        return self.data["s_area"]
+        return self._data["s_area"]
 
     @property
     def fanspeed(self) -> ViomiVacuumSpeed:
         """Current fan speed."""
-        return ViomiVacuumSpeed(self.data["suction_grade"])
+        return ViomiVacuumSpeed(self._data["suction_grade"])
 
     @property
     def water_grade(self) -> ViomiWaterGrade:
         """Water grade."""
-        return ViomiWaterGrade(self.data["water_grade"])
+        return ViomiWaterGrade(self._data["water_grade"])
 
     @property
     def remember_map(self) -> bool:
         """True to remember the map."""
-        return bool(self.data["remember_map"])
+        return bool(self._data["remember_map"])
 
     @property
     def has_map(self) -> bool:
         """True if device has map?"""
-        return bool(self.data["has_map"])
+        return bool(self._data["has_map"])
 
     @property
     def has_new_map(self) -> bool:
         """TODO: unknown"""
-        return bool(self.data["has_newmap"])
+        return bool(self._data["has_newmap"])
 
     @property
     def mop_mode(self) -> ViomiMode:
         """Whether mopping is enabled and if so which mode
 
         TODO: is this really the same as mode?
         """
-        return ViomiMode(self.data["is_mop"])
+        return ViomiMode(self._data["is_mop"])
+
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self._data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self._data["hw_info"]
+
+    @property
+    def sw_info(self) -> str:
+        """SoftWare info."""
+        return self._data["sw_info"]
+
+    @property
+    def hypa_hours_left(self) -> int:
+        """HYPA left hours."""
+        return self._data["hypa_hours"]
+
+    @property
+    def hypa_life_left(self) -> int:
+        """HYPA left life percent."""
```

Comment:
Is this just a misnamed HEPA filter, or something else?

Reply:
Yes, it's the HEPA filter, but the prop is really 'hypa_life'.
What should I do ? keep `hypa_life_left` or change it for `hepa_life_left`

Reply:
It is unfortunately not uncommon that the property names do not describe what the value really is. As this library builds an API for other people to use, I prefer reasonable naming over following the naming schemes used by device manufacturers.

So I think the name should be `hepa_filter_left` to follow the naming scheme from vacuumcontainers.py, even though (based on a grep through the code base) in some cases `life` is there in the name. So that part is not a big deal, pick whichever you think is more useful for any users of the lib.

Reply:
I renamed it to hepa_life_left, is it correct ?

Reply:
So I will rename it `hepa_filter_left`
Thanks !

BTW, do you how can I have the details for each function listed here:
* https://github.com/rytilahti/python-miio/issues/550#issuecomment-552780952
and here
* https://github.com/rytilahti/python-miio/issues/550#issuecomment-570808184

For example, the function, `rename_map` I don't know which parameter it takes ...

Reply:
Alas, capturing the necessary parameters require reverse engineering the plugin for that device, either by reading reverse-engineered code or capturing the traffic between the device and the official app. @rezmus may have an idea of the parameters you are looking for :-)

Reply:
Thanks ! I'm open the android plugin and I got what I need, I will try to add all the functions that I can :)

Reply:
done